site/src/pages/WorkspacePage/WorkspaceScheduleControls.tsx

```diff
@@ -0,0 +1,387 @@
+import { css } from "@emotion/css";
+import { type Interpolation, type Theme } from "@emotion/react";
+import Link, { LinkProps } from "@mui/material/Link";
+import { forwardRef, type FC } from "react";
+import { Link as RouterLink } from "react-router-dom";
+import { isWorkspaceOn } from "utils/workspace";
+import type { Workspace } from "api/typesGenerated";
+import {
+  autostartDisplay,
+  autostopDisplay,
+  getDeadline,
+  getMaxDeadline,
+  getMaxDeadlineChange,
+  getMinDeadline,
+} from "utils/schedule";
+import IconButton from "@mui/material/IconButton";
+import RemoveIcon from "@mui/icons-material/RemoveOutlined";
+import AddIcon from "@mui/icons-material/AddOutlined";
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+  usePopover,
+} from "components/Popover/Popover";
+import Tooltip from "@mui/material/Tooltip";
+import _ from "lodash";
+import { getErrorMessage } from "api/errors";
+import { decreaseDeadline, increaseDeadline } from "api/queries/workspaces";
+import { displaySuccess, displayError } from "components/GlobalSnackbar/utils";
+import { useMutation } from "react-query";
+
+export interface WorkspaceScheduleControlsProps {
+  workspace: Workspace;
+  canUpdateSchedule: boolean;
+}
+
+export const WorkspaceScheduleControls: FC<WorkspaceScheduleControlsProps> = ({
+  workspace,
+  canUpdateSchedule,
+}) => {
+  const deadline = getDeadline(workspace);
+  const maxDeadlineDecrease = getMaxDeadlineChange(deadline, getMinDeadline());
+  const maxDeadlineIncrease = getMaxDeadlineChange(
+    getMaxDeadline(workspace),
+    deadline,
+  );
+  const deadlinePlusEnabled = maxDeadlineIncrease >= 1;
+  const deadlineMinusEnabled = maxDeadlineDecrease >= 1;
```

Comment:
Do we have a policy around deadlines? As in:
1. Will it always be a value that's relative to the current time?
2. And once a deadline is up, is it not able to be interacted with any way?

Mainly curious because I would've expected the conditions for each of the derived values to be different

Reply:
Also, since it looks like these properties are only being used once each, do you think it'd be better to move them into the JSX output directly?

Reply:
About your questions:
1. Yes
2. Not sure if I got it. Can you say a use case? 

Honestly, I don't have too much context on this part and I have the same questions as you. @johnstcn probably can help us to understand this better.

Reply:
About JSX, I think I can better understand what it does by assigning this condition to a named variable.

Reply:
> Will it always be a value that's relative to the current time?

Yeah workspace deadline should generally be in the future. I could imagine rare situations where it's a few seconds in the past but those should be the exception.

> And once a deadline is up, is it not able to be interacted with any way?

There _might_ be a short window between the deadline expiring and the workspace actually being shut down in which you could theoretically still extend the deadline. But in general, I think it's fair to assume that once the deadline is up, workspace is stopping Very Soon.